hawkular-command-gateway-api/src/main/java/org/hawkular/cmdgw/api/UiSessionOrigin.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.api;
+
+import org.hawkular.bus.common.BasicMessage;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public interface UiSessionOrigin extends BasicMessage {
+    String getSenderRequestId();
```

Comment:
An interesting topic :) I am a proponent of making classes immutable by default. Here on messages that can perhaps get processed asynchronously, the immutability would make especially good sense. I had no time so far to have a look how and if at all the jsonschema2pojo-maven-plugin supports immutability, but to make a potential switch to immutable message classes easier, I am adding only those setters that I need in the server code.